src/backend/cdb/cdbllize.c

```diff
@@ -741,6 +750,15 @@ ParallelizeSubplan(SubPlan *spExpr, PlanProfile *context)
 		if (containingPlanDistributed)
 		{
 			Assert(NULL != context->currentPlanFlow);
+
+			if ((newPlan->flow->locustype == CdbLocusType_SegmentGeneral) &&
+				(contain_volatile_functions((Node *) newPlan->targetlist) ||
+				contain_volatile_functions(newPlan->qual)))
```

Comment:
Please provide explanation, why should we check only target list and quals for correlated queries?

Reply:
Moreover, there are several build warnings:

```
[2022-09-06T10:13:28.235Z] [0m[91mcreateplan.c: In function 'adjust_modifytable_flow':
[2022-09-06T10:13:28.235Z] [0m[91mcreateplan.c:6694:7: warning: passing argument 1 of 'contain_volatile_functions' from incompatible pointer type [enabled by default]
[2022-09-06T10:13:28.235Z]        contain_volatile_functions(subplan->qual))
[2022-09-06T10:13:28.235Z]        ^
[2022-09-06T10:13:28.235Z] [0m[91mIn file included from createplan.c:35:0:
[2022-09-06T10:13:28.235Z] ../../../../src/include/optimizer/clauses.h:86:13: note: expected 'struct Node *' but argument is of type 'struct List *'
[2022-09-06T10:13:28.235Z]  extern bool contain_volatile_functions(Node *clause);
[2022-09-06T10:13:28.235Z]              ^
[2022-09-06T10:13:28.235Z] [0m[91mcreateplan.c:6725:6: warning: passing argument 1 of 'contain_volatile_functions' from incompatible pointer type [enabled by default]
[2022-09-06T10:13:28.235Z]       !contain_volatile_functions(subplan->qual))
[2022-09-06T10:13:28.235Z]       ^
[2022-09-06T10:13:28.235Z] [0m[91mIn file included from createplan.c:35:0:
[2022-09-06T10:13:28.235Z] ../../../../src/include/optimizer/clauses.h:86:13: note: expected 'struct Node *' but argument is of type 'struct List *'
[2022-09-06T10:13:28.235Z]  extern bool contain_volatile_functions(Node *clause);
[2022-09-06T10:13:28.235Z]              ^
[2022-09-06T10:13:50.974Z] [0m[91mcdbllize.c: In function 'ParallelizeCorrelatedSubPlanMutator':
[2022-09-06T10:13:50.974Z] [0m[91mcdbllize.c:579:5: warning: passing argument 1 of 'contain_volatile_functions' from incompatible pointer type [enabled by default]
[2022-09-06T10:13:50.974Z]      contain_volatile_functions(scanPlan->qual)))
[2022-09-06T10:13:50.974Z]      ^
[2022-09-06T10:13:50.974Z] [0m[91mIn file included from cdbllize.c:18:0:
[2022-09-06T10:13:50.974Z] ../../../src/include/optimizer/clauses.h:86:13: note: expected 'struct Node *' but argument is of type 'struct List *'
[2022-09-06T10:13:50.974Z]  extern bool contain_volatile_functions(Node *clause);
[2022-09-06T10:13:50.974Z]              ^
[2022-09-06T10:13:50.974Z] [0m[91mcdbllize.c: In function 'ParallelizeSubplan':
[2022-09-06T10:13:50.974Z] [0m[91mcdbllize.c:756:5: warning: passing argument 1 of 'contain_volatile_functions' from incompatible pointer type [enabled by default]
[2022-09-06T10:13:50.974Z]      contain_volatile_functions(newPlan->qual)))
[2022-09-06T10:13:50.974Z]      ^
[2022-09-06T10:13:50.974Z] [0m[91mIn file included from cdbllize.c:18:0:
[2022-09-06T10:13:50.974Z] ../../../src/include/optimizer/clauses.h:86:13: note: expected 'struct Node *' but argument is of type 'struct List *'
[2022-09-06T10:13:50.974Z]  extern bool contain_volatile_functions(Node *clause);
[2022-09-06T10:13:50.974Z]              ^
```

take a look at CI output

Reply:
> there are several build warnings

fixed

Reply:
> Please provide explanation, why should we check only target list and quals for correlated queries?

change to target list and having quals with examples for both